File: package/main/src/String/padEnd.ts
/**
 * Adds the specified string to the end of the string until it reaches the specified length.
 *
 * @param string_ - The original string to apply padding
 * @param targetLength - The target length after padding
 * @param padString - The string to use for padding
 * @returns The string after padding has been applied
 * @throws {Error} If padString is empty
 */
export const padEnd = (
  string_: string,
  targetLength: number,
  padString: string,
): string => {
  if (padString === "") {
    throw new Error("padString cannot be empty");
  }
  let result = string_;
  while (result.length < targetLength) {
    result += padString.slice(0, targetLength - result.length);
  }
  return result;
};

File: package/umt_python/src/pad_end.py
def pad_end(string_: str, target_length: int, pad_string: str) -> str:
    """
    Adds the specified string to the end of the string until it reaches the specified length.

    Args:
        string_: The original string to apply padding.
        target_length: The target length after padding.
        pad_string: The string to use for padding.

    Returns:
        The string after padding has been applied.

    Example:
        >>> pad_end("abc", 5, "0")
        'abc00'
        >>> pad_end("hello", 8, "123")
        'hello123'
        >>> pad_end("world", 3, "!") # target_length is less than string length
        'world'
        >>> pad_end("test", 6, "") # pad_string is empty
        'test'
    """
    if not pad_string:
        return string_
    result = string_
    while len(result) < target_length:
        needed = target_length - len(result)
        result += pad_string[:needed]
    return result
